### PR TITLE
Savings module scaffold

### PR DESCRIPTION
**Pull request by denalimarsh**
Scaffold of the savings module

### CODE REVIEW

File: x/savings/keeper/keeper.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	"github.com/tendermint/tendermint/libs/log"
+
+	"github.com/kava-labs/kava/x/savings/types"
+)
+
+// Keeper struct for pricefeed module
```
Comment (DracoLi) [inline x/savings/keeper/keeper.go:14]: looks like some comments needs to be updated

Comment (denalimarsh) [reply x/savings/keeper/keeper.go]: fixed in https://github.com/Kava-Labs/kava/pull/1188/commits/3d2082e21d168df30ca4105b160bc9b4378b1c08

File: proto/kava/savings/v1beta1/query.proto [hunk 2]
```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+package kava.savings.v1beta1;
+
+import "kava/savings/v1beta1/store.proto";
+import "gogoproto/gogo.proto";
+import "google/api/annotations.proto";
+import "google/protobuf/timestamp.proto";
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply proto/kava/savings/v1beta1/query.proto]: Updated in https://github.com/Kava-Labs/kava/pull/1188/commits/ed838796330f433690e76b424bc16339308f0ea8

**Review by drklee3 — APPROVED** [review]
lgtm, just the legacy stuff (queryhandler, rest client) I don't think we don't need since its a new module that doesn't need to be backwards compatible

File: x/savings/abci.go [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package savings
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/kava-labs/kava/x/savings/keeper"
+)
+
+// EndBlocker updates the current pricefeed
```
Comment (karzak) [inline x/savings/abci.go:8]: pricefeed reference

File: x/savings/types/codec.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/codec/types"
+
+	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// RegisterLegacyAminoCodec registers all the necessary types and interfaces for the
+// governance module.
```
Comment (karzak) [inline x/savings/types/codec.go:13]: governance module reference

File: x/savings/types/params.go [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+// Parameter keys
+var ()
+
+// NewParams creates a new Params object
+func NewParams() Params {
+	return Params{}
+}
+
+// DefaultParams default params for pricefeed
```
Comment (karzak) [inline x/savings/types/params.go:15]: pricefeed reference

File: x/savings/types/params.go [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+	return NewParams()
+}
+
+// ParamKeyTable Key declaration for parameters
+func ParamKeyTable() paramtypes.KeyTable {
+	return paramtypes.NewKeyTable().RegisterParamSet(&Params{})
+}
+
+// ParamSetPairs implements the ParamSet interface and returns all the key/value pairs
+// pairs of pricefeed module's parameters.
```
Comment (karzak) [inline x/savings/types/params.go:26]: ditto

**Review by karzak — CHANGES_REQUESTED** [review]
Let's add the `savings` module account as part of this PR - doesn't need any additional permissions

**Comment by rhuairahrighairidh** [comment]
> lgtm, just the legacy stuff (queryhandler, rest client) I don't think we don't need since its a new module that doesn't need to be backwards compatible

+1 `keeper/querier.go` and `client/rest` won't be needed

File: x/savings/abci.go (hunk 3 above)
Comment (rhuairahrighairidh) [reply x/savings/abci.go:8]: We don't need an endblocker for this module right? If so lets delete the file.

Comment (denalimarsh) [reply x/savings/abci.go]: Followed @rhuairahrighairidh's suggestion and removed abci in https://github.com/Kava-Labs/kava/pull/1188/commits/2d3046e3226ae6038c686b53dc3f3be355a00b0b

File: x/savings/types/params.go (hunk 5 above)
Comment (denalimarsh) [reply x/savings/types/params.go]: https://github.com/Kava-Labs/kava/pull/1188/commits/259d84e242aff81d4cd0665ab99ae7a8b14745b3

File: x/savings/types/params.go (hunk 6 above)
Comment (denalimarsh) [reply x/savings/types/params.go]: https://github.com/Kava-Labs/kava/pull/1188/commits/259d84e242aff81d4cd0665ab99ae7a8b14745b3

File: x/savings/types/codec.go (hunk 4 above)
Comment (denalimarsh) [reply x/savings/types/codec.go]: - [ ] https://github.com/Kava-Labs/kava/pull/1188/commits/259d84e242aff81d4cd0665ab99ae7a8b14745b3

**Comment by denalimarsh** [comment]
> Let's add the `savings` module account as part of this PR - doesn't need any additional permissions

Think this was already done in app.go. I added a genesis_test in https://github.com/Kava-Labs/kava/pull/1188/commits/a0374ef4fec3b0681f54b65cd8fa8989dbd3bd9d to confirm module account initialization. Are there additional changes needed that are missing?

**Comment by denalimarsh** [comment]
> > lgtm, just the legacy stuff (queryhandler, rest client) I don't think we don't need since its a new module that doesn't need to be backwards compatible
> 
> +1 `keeper/querier.go`, `types/querier.go` and `client/rest` won't be needed

Noted. Let's leave as is for now and handle this as part of the querier work in https://github.com/Kava-Labs/kava/pull/1193.

**Review by karzak — APPROVED** [review]

**Pull request merged**